app/routes/questions/add.tsx

```diff
@@ -0,0 +1,74 @@
+import {useState} from 'react'
+import type { ActionArgs } from '@remix-run/cloudflare'
+import { useActionData, Form, useSearchParams, useTransition, useSubmit } from '@remix-run/react'
+import { redirect } from '@remix-run/cloudflare'
+import { addQuestion, loadAllQuestions } from '~/server-utils/stampy'
+
+
+export const action = async ({ request }: ActionArgs) => {
+    const formData = await request.formData();
+    let title = formData.get('title')
+    const state = formData.get('stateString');
+    const redirectTo = '/' + (state ? '?state=' + state : '')
+    const relatedQuestions = formData.getAll('relatedQuestion').map(question => ({ title: question }));
+
+    // Make sure that the question was provided
+    if (!title) return redirect(redirectTo);
+
+    // Check whether the question is simply a prefix of an existant question,
+    // and if so ignore it
+    const allQuestions = await loadAllQuestions(request);
+    const isPrefix = allQuestions.data.some(
+        question => question.toLowerCase().startsWith(title.toLowerCase())
+    );
+    if (isPrefix) return redirect(redirectTo);
```

Comment:
I'm not sure about this. It will handle things like "Why might a superintelligent AI be da", but e.g. also remove questions that are valid subsets of existing ones, like "What is AI" vs "What is AI safety"

Reply:
whatever you decide is the desired behaviour in discussion with @plexish about all the branches of this code...

I don't understand all these redirects and haven't seen written specification for this feature, but I don't remember anything about redirects myself 🤔

Reply:
Remix tries to gracefully fallback when there is no JS. Which means that action function have to return valid HTTP. This endpoint is `questions/add`, so it needs to make sure that the page stays at `http://127.0.0.1:8787/` after the question is added, but a form submit pretty much is a page reload. So this function needs to both redirect to the root site, but also to make sure that any selected questions (i.e. the state query param) stay selected.
That's a long winded way of saying that you should just think of `redirect(redirectTo)` as a fancy `return`

Reply:
ah, ok then 👍

Reply:
> "What is AI" vs "What is AI safety"
hm, this seems like a small enough set of edge cases that's it's not too bad, but also adding substrings is not a big problem. Not a strong preference here, whichever is easy/already done seems fine.